pulp_2to3_migration/app/pre_migration.py

```diff
@@ -457,16 +461,34 @@ def pre_migrate_distributor(repo_id, distributors, distributor_types, repo=None)
         last_updated = (dist_data.last_updated
                         and timezone.make_aware(dist_data.last_updated, timezone.utc))
 
-        Pulp2Distributor.objects.create(
+        distributor, created = Pulp2Distributor.objects.get_or_create(
             pulp2_object_id=dist_data.id,
-            pulp2_id=dist_data.distributor_id,
-            pulp2_type_id=dist_data.distributor_type_id,
-            pulp2_last_updated=last_updated,
-            pulp2_config=dist_data.config,
-            pulp2_repository=repo,
-            pulp2_repo_id=repo_id,
-            is_migrated=False
-        )
+            defaults={'pulp2_id': dist_data.distributor_id,
+                      'pulp2_type_id': dist_data.distributor_type_id,
+                      'pulp2_last_updated': last_updated,
+                      'pulp2_config': dist_data.config,
+                      'pulp2_repository': repo,
+                      'pulp2_repo_id': repo_id,
+                      'is_migrated': False})
+
+        if not created:
+            # if it was marked as such because it was not present in the migration plan
+            distributor.not_in_plan = False
+
+            if last_updated != distributor.pulp2_last_updated:
+                distributor.pulp2_config = dist_data.config
+                dist_migrator = distributor_migrators.get(distributor.pulp2_type_id)
+                needs_new_publication = dist_migrator.needs_new_publication(distributor)
+                if needs_new_publication and distributor.pulp3_publication:
+                    distributor.pulp3_publication.delete()
+                else:
+                    needs_new_distribution = dist_migrator.needs_new_distribution(distributor)
+                    if needs_new_distribution and distributor.pulp3_distribution:
+                        distributor.pulp3_distribution.delete()
+
+                distributor.pulp2_last_updated = last_updated
```

Comment:
we should update old config with the new one
distributor.pulp2_config = dist_data.config

Reply:
nevermind, i see it on line 479, maybe move all field updates in one place for better readability?

Reply:
Yeah, I moved the config up because it is used in the calls right after it.
Thanks, I moved everything up.